### PR TITLE
Migration for fixing author analytics

### PR DESCRIPTION
**Pull request by Will-Howard**
Because aggregating the analytics data is quite slow/fragile, I would like to deploy this migration to prod so I can test https://github.com/ForumMagnum/ForumMagnum/pull/8640 with the prod data to try and catch any bugs before deploying that one

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1206361219156345) by [Unito](https://www.unito.io)

### CODE REVIEW

**Comment by jpaddison3** [comment]
Why is the new table in the main DB?

**Comment by Will-Howard** [comment]
> Why is the new table in the main DB?

@jpaddison3 Some reasons I think it's better this way:
- We can join on other tables if needed (e.g. on Posts to make the reads metric incorporate the expected reading time of the post)
- Really the only reason it was in the analytics db before was because it had to be to take advantage of the built in materialized view thing. Without that requirement, I think it makes sense to put it in the main db for the same reason we would put any other table in the main db (it's simpler to just have one db to look at most of the time, higher + more predictable performance, infrastructure for defining a schema + indexes already exists)

**Comment by jpaddison3** [comment]
> Really the only reason it was in the analytics db before

The reason we have the analytics DB in the first place is so we can hammer it with analytics queries without risking hurting the performance of the main site. I would have thought that was the main reason.

> as noted in https://github.com/ForumMagnum/ForumMagnum/pull/8640

Sorry I'm dumb, where?

**Comment by Will-Howard** [comment]
> The reason we have the analytics DB in the first place is so we can hammer it with analytics queries without risking hurting the performance of the main site. I would have thought that was the main reason.

Hmm this is reasonable. I just checked how big the two new tables will be to try and get an idea of how much of a problem they might cause:
- PostViewTimes will be ~4GB, 8M rows
- PostViews will be ~1GB, 3M rows

And then for comparison, the `raw` analytics table is ~400GB, 600M rows. And here are the biggest tables on the prod db:

| Table Name           | Size    | Row Count  |
|----------------------|---------|------------|
| ClientIds            | 37 GB   | 127,887,033|
| LWEvents             | 33 GB   | 52,629,002 |
| Sessions             | 6235 MB | 8,635,154  |
| Revisions            | 4217 MB | 453,924    |
| Votes                | 2209 MB | 2,349,212  |
| PostRecommendations  | 1882 MB | 2,900,606  |
| ReadStatuses         | 1422 MB | 2,884,035  |
| Posts                | 1392 MB | 39,079     |
| Comments             | 1176 MB | 161,887    |
| PostEmbeddings       | 816 MB  | 39,055     |
| Users                | 585 MB  | 38,017     |


> Sorry I'm dumb, where?

File: packages/lesswrong/lib/collections/postViewTimes/collection.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { createCollection } from '../../vulcan-lib';
+import { addUniversalFields, getDefaultResolvers } from '../../collectionUtils'
+import { getDefaultMutations } from '../../vulcan-core/default_mutations';
+import { schema } from './schema';
+import { ensureIndex } from '../../collectionIndexUtils';
+
+export const PostViewTimes = createCollection({
```
Comment (jpaddison3) [inline packages/lesswrong/lib/collections/postViewTimes/collection.ts:7]: Please add documentation for what this collection is, how it's used, and why it exists.

File: packages/lesswrong/lib/collections/postViews/collection.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { createCollection } from '../../vulcan-lib';
+import { addUniversalFields, getDefaultResolvers } from '../../collectionUtils'
+import { getDefaultMutations } from '../../vulcan-core/default_mutations';
+import { schema } from './schema';
+import { ensureIndex } from '../../collectionIndexUtils';
+
+export const PostViews = createCollection({
```
Comment (jpaddison3) [inline packages/lesswrong/lib/collections/postViews/collection.ts:7]: Documentation here as well.

File: packages/lesswrong/lib/collections/postViewTimes/schema.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import { foreignKeyField } from "../../utils/schemaUtils";
+
+export const schema: SchemaType<"PostViewTimes"> = {
```
Comment (jpaddison3) [inline packages/lesswrong/lib/collections/postViewTimes/schema.ts:3]: I would love some documentation of the fields as well, but I'll leave that to you.

**Review by jpaddison3 — APPROVED** [review]
Sorry for being slow here. I think I'm ok with it being on the main db. I think I'd still find it more natural if it were on the analytics db, but not strongly.

I've made some requests for documentation. Approved subject to those.

**Pull request merged**